Orange/widgets/data/owmergedata.py

```diff
@@ -369,8 +369,12 @@ def _join_table_by_indices(self, reduced_extra, indices):
         string_cols = [i for i, var in enumerate(domain.metas) if var.is_string]
         metas = self._join_array_by_indices(
             self.data.metas, reduced_extra.metas, indices, string_cols)
-        return Orange.data.Table.from_numpy(domain, X, Y, metas)
-
+        table = Orange.data.Table.from_numpy(domain, X, Y, metas)
+        table.name = getattr(self.data, 'name')
+        table.attributes = getattr(self.data, 'attributes')
+        table.ids = getattr(self.data, 'ids')
+        return table 
```

Comment:
I don't know this code well, but judging by what I see in the `Table` class, I think that you should have `getattr(self.data, 'name', '')` and `getattr(self.data, 'attributes', {})`.

`ids` seems to be always present, so it's just `table.ids = self.data.ids`.

(Perhaps `name` and `attributes` are always present, too, but a lot of code does `getattr` for them.)